rust/src/sip/sip.rs

```diff
@@ -112,6 +109,15 @@ impl SIPState {
         }
     }
 
+    fn append_request(&mut self, input: &[u8], request: Request) {
+        let mut tx = self.new_tx(crate::core::Direction::ToServer);
+        tx.request = Some(request);
+        if let Ok((_, req_line)) = sip_take_line(input) {
+            tx.request_line = req_line;
+        }
+        self.transactions.push(tx);
```

Comment:
Maybe we should put a limit on the maximum number of transactions...

Reply:
With this parse, the existence of the transaction means complete. Are we running into max size issues with such protocols that don't correlate requests and responses before considering the tx complete?

Reply:
Right, this is more needed for protocols where transaction can outlive the packet where they were created